### ISSUE TITLE
Bug - Filter called only once per URL

### ISSUE THREAD

**Issue by mianor64** [awslabs/aws-serverless-java-container#56]
**Summary**
Filters are not working as expected (at least for SpringLambdaContainerHandler). Filters are called only the first time a URL is called. I used the sample pet-store for Spring and added a filter as described in the documentation.
First time I called an URL (i.e. /pets/1), the filter is executed as expected. If I call the same URL again, the filter is not executed.
If I call another URL, (i.e. /pets/2), the filter is executed the first time, but not executed for the subsequent calls to that same URL.

**Steps to reproduce**
From aws-serverless-java-container-master\samples\spring\pet-store, modify LambdaHandler.java to add a filter, named TestFilter which will only logged (into CloudWatch) when the doFilter method is called (see in attachments).

`try {`
`handler = SpringLambdaContainerHandler.getAwsProxyHandler(PetStoreSpringAppConfig.class);`
`handler.onStartup(c -> {`
`FilterRegistration.Dynamic registration = c.addFilter("TestFilter", TestFilter.class);`
`registration.addMappingForUrlPatterns(EnumSet.of(DispatcherType.REQUEST), true, "/*");`
`});`
`} catch (ContainerInitializationException e) {`

Add the TestFilter class (see in attachments):

`public class TestFilter implements Filter {`
`    @Override`
`    public void doFilter(final ServletRequest req, final ServletResponse res, final FilterChain chain) throws IOException, ServletException {`
`        System.out.println("DoFilter TestFilter");`
`        chain.doFilter(req, res);`
`    }`

**Expected Result**
I expected the TestFilter doFilter() method to be executed each time I access any URL

**Actual Result**
First time I called an URL (i.e. /pets/1), the filter is executed as expected. If I call the same URL again, the filter is not executed.
If I call another URL, (i.e. /pets/2), the filter is executed the first time, but not executed for the subsequent calls to that same URL.

[aws-serverless-spring-sample-with-filter.zip](https://github.com/awslabs/aws-serverless-java-container/files/1220501/aws-serverless-spring-sample-with-filter.zip)

**Comment by SAPessi** [comment]
Thanks @mianor64 - I'll investigate and push a fix to 0.7-SNAPSHOT next week

**Comment by SAPessi** [comment]
Hi @mianor64, I've found the issue and addressed it in the last commit on the `servlet-improvements` branch. I've also added a unit test for this case. If you don't mind, could you test with your application:

1. Clone the `servlet-improvements` branch from GitHub:
```bash
$ git clone -b servlet-improvements https://github.com/awslabs/aws-serverless-java-container.git
```

2. Use maven to install the new version in local:
```bash
$ mvn install 
```

3. Update your pom.xml to use version `0.7-SNAPSHOT`

**Comment by mianor64** [comment]
Hi @SAPessi, I confirm it's working now! Thanks for your fast answer.

**Comment by SAPessi** [comment]
Great, thanks for checking @mianor64, I'll close this and merge into master today.
